server/src/main/java/org/elasticsearch/common/geo/EdgeTreeReader.java

```diff
@@ -33,16 +33,21 @@
     private final ByteBufferStreamInput input;
     private final int startPosition;
     private final boolean hasArea;
+    private Extent treeExtent;
 
     public EdgeTreeReader(ByteBufferStreamInput input, boolean hasArea) throws IOException {
         this.startPosition = input.position();
         this.input = input;
         this.hasArea = hasArea;
+        this.treeExtent = null;
     }
 
     public Extent getExtent() throws IOException {
         resetInputPosition();
-        return new Extent(input);
+        if (treeExtent == null) {
```

Comment:
move resetInputPosition() inside the if condition?

Reply:
good call